### PR TITLE
SLS-592 Extend version cursor to walk across keys

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

**Review by korteland — APPROVED** [review]
LGTM

File: test/suite/test_cursor18.py [hunk 1]
```diff
@@ -558,3 +558,54 @@ def test_skip_tombstone_on_disk(self):
         self.assertEquals(version_cursor.get_key(), 1)
         self.verify_value(version_cursor, 1, 1, WT_TS_MAX, WT_TS_MAX, 3, 0, 0, 1, 0)
         self.assertEquals(version_cursor.next(), wiredtiger.WT_NOTFOUND)
+
+    def test_multiple_keys_with_search(self):
+        self.create()
```
Comment (ddanderson) [inline test/suite/test_cursor18.py:563]: Is there a way to get the old version cursor behavior - that is, just iterate through the values for a single key?  If not, that's okay - but does doc need to be updated?

**Review by ddanderson — APPROVED** [review]
LGTM - a question and a nit.

File: test/suite/test_cursor18.py (hunk 1 above)
Comment (quchenhao) [reply test/suite/test_cursor18.py:563]: Yes, you are right. The behaviour will change. I'll update the doc if there is any.

File: test/suite/test_cursor18.py [hunk 2]
```diff
@@ -558,3 +558,54 @@ def test_skip_tombstone_on_disk(self):
...
+        self.session.commit_transaction("commit_timestamp=" + self.timestamp_str(5))
+
+        # Open a version cursor
+        self.session.begin_transaction()
+        version_cursor = self.session.open_cursor(self.uri, None, "debug=(dump_version=(enabled=true))")
+        version_cursor.set_key(1)
+        self.assertEquals(version_cursor.search(), 0)
+        self.assertEquals(version_cursor.get_key(), 1)
+        self.verify_value(version_cursor, 1, 1, WT_TS_MAX, WT_TS_MAX, 3, 0, 0, 0, 0)
+        self.assertEquals(version_cursor.next(), 0)
```
[the remark this answers is not in the document]
Comment (quchenhao) [reply test/suite/test_cursor18.py:583]: This intention of the test is for the scenario that we do a search and then next.

**Pull request merged**